application_form/models.py

```diff
@@ -1,217 +1,25 @@
-            raise ValueError(
-                _("the offer cannot be accepted before the application is approved.")
-            )
-        super(ApplicationMixin, self).clean()
-
-    def save(self, **kwargs):
-        self.full_clean()
-        return super(ApplicationMixin, self).save(**kwargs)
-
-    def reject(self, rejection_description: str) -> None:
-        self.is_approved = False
-        self.is_rejected = True
-        self.rejection_description = rejection_description
-        self._change_reason = _("application rejected.")
-        self.save()
-
-    def approve(self) -> None:
-        self.is_approved = True
-        self._change_reason = _("application approved.")
-        self.save()
-
-    def accept_offer(self, apartment: Apartment) -> None:
-        from application_form.services import accept_offer
-
-        accept_offer(self, apartment)
-
-
-class HasoApplication(ApplicationMixin):
-    right_of_occupancy_id = models.IntegerField(verbose_name=_("right of occupancy ID"))
-    current_housing = models.CharField(
-        max_length=255,
-        choices=CURRENT_HOUSING_CHOICES,
-        verbose_name=_("current housing"),
-    )
-    housing_description = models.TextField(verbose_name=_("housing description"))
-    housing_type = models.CharField(max_length=255, verbose_name=_("housing type"))
-    housing_area = models.FloatField(verbose_name=_("housing area"))
-    is_changing_occupancy_apartment = models.BooleanField(
-        default=False, verbose_name=_("is changing occupancy apartment")
-    )
-    is_over_55 = models.BooleanField(
-        default=False, verbose_name=_("is applicant over 55 years old")
-    )
-
-    @property
-    def apartment_uuids(self):
-        from application_form.selectors import get_haso_apartment_uuids
-
-        return get_haso_apartment_uuids(self)
-
-
-class HasoApartmentPriority(models.Model):
-    is_active = models.BooleanField(default=True, verbose_name=_("is active"))
-    priority_number = models.IntegerField(verbose_name=_("priority number"))
-    haso_application = models.ForeignKey(
-        HasoApplication,
-        on_delete=models.CASCADE,
-        related_name="haso_apartment_priorities",
-    )
-    apartment = models.ForeignKey(
-        Apartment,
-        on_delete=models.CASCADE,
-        related_name="haso_apartment_priorities",
-    )
-    history = HistoricalRecords()
-
-    class Meta:
-        permissions = HASO_PERMISSIONS_LIST
-
-
-class HitasApplication(ApplicationMixin):
-    has_previous_hitas_apartment = models.BooleanField(
-        default=False, verbose_name=_("has previous hitas apartment")
-    )
-    previous_hitas_description = models.TextField(
-        verbose_name=_("previous hitas descripiton")
-    )
-    has_children = models.BooleanField(default=False, verbose_name=_("has children"))
-    apartment = models.ForeignKey(
-        Apartment, on_delete=models.CASCADE, related_name="hitas_applications"
+class Application(models.Model):
```

Comment:
Looking from model description, i think there is going to be application id coming from drupal.

Reply:
Confirmed that it will be UUID field

Reply:
Added the missing field.